### PR TITLE
Build pull requests and master with GitHub Actions

### PR DESCRIPTION
**Pull request by quantum5**
After this PR is merged, on every subsequent PR it should say whether the PR builds successfully or not.

Currently, the following components are built:
* [x] client
* [x] kvmfr kernel module
* [x] linux host
* [x] windows host (cross compile)
* [x] obs plugin

Note that in 9ff1859dc134e9f3196703296f868040c77a4dd6, `ivshmemInit` and `ivshmemFree` were split from `ivshmemOpen` and `ivshmemClose`, but only for Windows. To fix the build for the linux host, I created stubs, but the same split should be done at a later time.

### CODE REVIEW

**Comment by JJRcop** [comment]
This should also install `makensis` and build the host installer program

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  host-linux:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: recursive
+    - name: Install dependencies
+      run: |
+        sudo apt-get install binutils-dev libgl1-mesa-dev
+    - name: Configure host
```
Comment (JJRcop) [inline .github/workflows/build.yml:45]: `Configure Linux host`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      run: |
+        mkdir host/build
+        cd host/build
+        cmake ..
+    - name: Build host
```
Comment (JJRcop) [inline .github/workflows/build.yml:50]: `Build Linux host`

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  host-windows:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: recursive
+    - name: Install dependencies
+      run: |
+        sudo apt-get install gcc-mingw-w64-x86-64 g++-mingw-w64-x86-64
+    - name: Configure host
```
Comment (JJRcop) [inline .github/workflows/build.yml:64]: `Configure Windows host`

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      run: |
+        mkdir host/build
+        cd host/build
+        cmake -DCMAKE_TOOLCHAIN_FILE=../toolchain-mingw64.cmake ..
+    - name: Build host
```
Comment (JJRcop) [inline .github/workflows/build.yml:69]: `Build Windows host`

**Review by JJRcop — CHANGES_REQUESTED** [review]
Differentiate between Windows and Linux hosts in action names. Right now they are ambiguous

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,91 @@
+name: build
+on: [push, pull_request]
+jobs:
+  client:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: recursive
+    - name: Install dependencies
```
Comment (JJRcop) [inline .github/workflows/build.yml:10]: `Install client dependencies`

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        cd module
+        make
+
+  host-linux:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: recursive
+    - name: Install dependencies
```
Comment (JJRcop) [inline .github/workflows/build.yml:42]: `Install Linux host dependencies`

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        cd host/build
+        make -j$(nproc)
+
+  host-windows:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: recursive
+    - name: Install dependencies
```
Comment (JJRcop) [inline .github/workflows/build.yml:61]: `Install Windows host dependencies`

File: .github/workflows/build.yml [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+        cd host/build
+        make -j$(nproc)
+
+  obs:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: recursive
+    - name: Install dependencies
```
Comment (JJRcop) [inline .github/workflows/build.yml:80]: `Install obs plugin dependencies`

**Review by JJRcop — CHANGES_REQUESTED** [review]
Ambiguous dependencies steps, same name each time

**Comment by gnif** [comment]
Agreed with @JJRcop's suggestions, please make these alterations after which I am happy to merge this

**Comment by quantum5** [comment]
Hmm currently doesn't build. Don't merge until investigated.

**Comment by quantum5** [comment]
Okay it should be fine now. The client failing to build is due to needing to update the submodule after gnif/PureSpice#5.

**Pull request merged**